diff --git a/python/private/pypi/package_annotation.bzl b/python/private/pypi/package_annotation.bzl
index 4a54703a..a2be9e67 100644
--- a/python/private/pypi/package_annotation.bzl
+++ b/python/private/pypi/package_annotation.bzl
@@ -20,7 +20,8 @@ def package_annotation(
         copy_executables = {},
         data = [],
         data_exclude_glob = [],
-        srcs_exclude_glob = []):
+        srcs_exclude_glob = [],
+        excluded_deps = []):
     """Annotations to apply to the BUILD file content from package generated from a `pip_repository` rule.
 
     [cf]: https://github.com/bazelbuild/bazel-skylib/blob/main/docs/copy_file_doc.md
@@ -35,6 +36,7 @@ def package_annotation(
         data_exclude_glob (list, optional): A list of exclude glob patterns to add as `data` to the generated
             `py_library` target.
         srcs_exclude_glob (list, optional): A list of labels to add as `srcs` to the generated `py_library` target.
+        deps (list, optional): A list of labels to remove as `deps` to the generated `py_library` target.
 
     Returns:
         str: A json encoded string of the provided content.
@@ -46,4 +48,5 @@ def package_annotation(
         data = data,
         data_exclude_glob = data_exclude_glob,
         srcs_exclude_glob = srcs_exclude_glob,
+        excluded_deps = excluded_deps,
     ))
