diff --git a/python/private/pypi/generate_whl_library_build_bazel.bzl b/python/private/pypi/generate_whl_library_build_bazel.bzl
index 8050cd22..6beed186 100644
--- a/python/private/pypi/generate_whl_library_build_bazel.bzl
+++ b/python/private/pypi/generate_whl_library_build_bazel.bzl
@@ -15,6 +15,7 @@
 """Generate the BUILD.bazel contents for a repo defined by a whl_library."""
 
 load("//python/private:text_util.bzl", "render")
+load("//python/private:normalize_name.bzl", "normalize_name")
 
 _RENDER = {
     "copy_executables": render.dict,
@@ -67,6 +68,14 @@ def generate_whl_library_build_bazel(
         if annotation.additive_build_content:
             additional_content.append(annotation.additive_build_content)
 
+        deps = []
+        exclude_deps = [normalize_name(d) for d in annotation.exclude_deps]
+        include_deps = [normalize_name(d) for d in kwargs["dependencies"]]
+        for d in include_deps:
+            if d not in exclude_deps:
+                deps += [d]
+        kwargs["dependencies"] = deps
+
     contents = "\n".join(
         [
             _TEMPLATE.format(
diff --git a/python/private/pypi/package_annotation.bzl b/python/private/pypi/package_annotation.bzl
index 4a54703a..692e19ee 100644
--- a/python/private/pypi/package_annotation.bzl
+++ b/python/private/pypi/package_annotation.bzl
@@ -20,7 +20,9 @@ def package_annotation(
         copy_executables = {},
         data = [],
         data_exclude_glob = [],
-        srcs_exclude_glob = []):
+        srcs_exclude_glob = [],
+        exclude_deps = [],
+    ):
     """Annotations to apply to the BUILD file content from package generated from a `pip_repository` rule.
 
     [cf]: https://github.com/bazelbuild/bazel-skylib/blob/main/docs/copy_file_doc.md
@@ -46,4 +48,5 @@ def package_annotation(
         data = data,
         data_exclude_glob = data_exclude_glob,
         srcs_exclude_glob = srcs_exclude_glob,
+        exclude_deps = exclude_deps,
     ))
