diff --git a/foreign_cc/private/cmake_script.bzl b/foreign_cc/private/cmake_script.bzl
index df697ab..31a155d 100644
--- a/foreign_cc/private/cmake_script.bzl
+++ b/foreign_cc/private/cmake_script.bzl
@@ -360,6 +360,7 @@ def _fill_crossfile_from_toolchain(workspace_name, tools, flags, target_os):
     #        lines += [_set_list(ctx, "CMAKE_STATIC_LINKER_FLAGS_INIT", flags.cxx_linker_static)]
     if flags.cxx_linker_shared:
         dict["CMAKE_SHARED_LINKER_FLAGS_INIT"] = _join_flags_list(workspace_name, flags.cxx_linker_shared)
+        dict["CMAKE_MODULE_LINKER_FLAGS_INIT"] = _join_flags_list(workspace_name, flags.cxx_linker_shared)
 
         # cxx_linker_shared will contain '-shared' or '-dynamiclib' on macos. This flag conflicts with "-bundle"
         # that is set by CMAKE based on platform. e.g.
diff --git a/foreign_cc/private/framework.bzl b/foreign_cc/private/framework.bzl
index e552f12..6d0f453 100644
--- a/foreign_cc/private/framework.bzl
+++ b/foreign_cc/private/framework.bzl
@@ -1104,6 +1104,9 @@ def _expand_locations(ctx, expandable, data):
     else:
         fail("Unsupported type: {}".format(type(expandable)))
 
+# ENFABRICA PATCH: export these functions for our own rules
+expand_locations = _expand_locations
+
 def _expand_locations_in_string(ctx, expandable, data):
     # If `EXT_BUILD_ROOT` exists in the string, we assume the user has added it themselves
     if "EXT_BUILD_ROOT" in expandable:
