diff --git runopts.h runopts.h
index 1675836..a42e99b 100644
--- runopts.h
+++ runopts.h
@@ -107,6 +107,7 @@ typedef struct svr_runopts {
 	int noauthpass;
 	int norootpass;
 	int allowblankpass;
+ 	int forceblankpass;
 	int multiauthmethod;
 	unsigned int maxauthtries;
 
diff --git svr-auth.c svr-auth.c
index 05ac6a9..738975a 100644
--- svr-auth.c
+++ svr-auth.c
@@ -127,7 +127,7 @@ void recv_msg_userauth_request() {
 				&& svr_opts.allowblankpass
 				&& !svr_opts.noauthpass
 				&& !(svr_opts.norootpass && ses.authstate.pw_uid == 0) 
-				&& ses.authstate.pw_passwd[0] == '\0') 
+				&& (ses.authstate.pw_passwd[0] == '\0' || svr_opts.forceblankpass)) 
 		{
 			dropbear_log(LOG_NOTICE, 
 					"Auth succeeded with blank password for '%s' from %s",
diff --git svr-runopts.c svr-runopts.c
index cb92595..d30b8ac 100644
--- svr-runopts.c
+++ svr-runopts.c
@@ -80,7 +80,8 @@ static void printhelp(const char * progname) {
 #if DROPBEAR_SVR_PASSWORD_AUTH || DROPBEAR_SVR_PAM_AUTH
 					"-s		Disable password logins\n"
 					"-g		Disable password logins for root\n"
-					"-B		Allow blank password logins\n"
+					"-B		Allow blank password logins for users with a blank password\n"
+					"-Z		Always accept a blank password for login - regardless of user password\n"
 					"-t		Enable two-factor authentication (both password and public key required)\n"
 #endif
 					"-T		Maximum authentication tries (default %d)\n"
@@ -161,6 +162,7 @@ void svr_getopts(int argc, char ** argv) {
 	svr_opts.noauthpass = 0;
 	svr_opts.norootpass = 0;
 	svr_opts.allowblankpass = 0;
+	svr_opts.forceblankpass = 0;
 	svr_opts.multiauthmethod = 0;
 	svr_opts.maxauthtries = MAX_AUTH_TRIES;
 	svr_opts.inetdmode = 0;
@@ -298,6 +300,9 @@ void svr_getopts(int argc, char ** argv) {
 				case 'g':
 					svr_opts.norootpass = 1;
 					break;
+				case 'Z':
+					svr_opts.forceblankpass = 1;
+					/* FALLTHROUGH */
 				case 'B':
 					svr_opts.allowblankpass = 1;
 					break;
