diff -urN b/BUILD.bazel c/BUILD.bazel
--- b/BUILD.bazel	1969-12-31 16:00:00
+++ c/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,6 @@
+# gazelle:go_naming_convention import_alias
+# gazelle:proto package
+# gazelle:proto_group go_package
+# gazelle:exclude gapic
+# gazelle:exclude third_party
+# gazelle:exclude google/example/endpointsapis/goapp
diff -urN b/google/BUILD.bazel c/google/BUILD.bazel
--- b/google/BUILD.bazel	1969-12-31 16:00:00
+++ c/google/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1 @@
+# gazelle:prefix google.golang.org/genproto/googleapis
