diff --git a/bin/bats b/bin/bats
index 1f7ce2e..269caf0 100755
--- a/bin/bats
+++ b/bin/bats
@@ -54,6 +54,7 @@ if ! BATS_PATH=$(bats_readlinkf "${BASH_SOURCE[0]}" 2>/dev/null); then
   BATS_PATH=$(bats_readlinkf "${BASH_SOURCE[0]}")
 fi
 
-export BATS_ROOT=${BATS_PATH%/*/*}
+export BATS_ROOT="${BATS_PATH%/bats}"
+export BATS_ROOT="${BATS_ROOT%/bin}"
 export -f bats_readlinkf
 exec env BATS_ROOT="$BATS_ROOT" "$BATS_ROOT/libexec/bats-core/bats" "$@"
