diff -Naur a/query.go b/query.go
--- a/query.go	2023-02-20 19:12:15.314113064 +0000
+++ b/query.go	2023-02-21 00:59:34.987308366 +0000
@@ -928,3 +928,9 @@
 
 // AggregationResult contains the results of an aggregation query.
 type AggregationResult map[string]interface{}
+
+// ENFABRICA PATCH
+// Export toProto for use in unit tests
+func (q *Query) ToProto(req *pb.RunQueryRequest) error {
+	return q.toRunQueryRequest(req)
+}
