diff --git a/query.go b/query.go
index f2e6d9c2a..1d6c69a6d 100644
--- a/query.go
+++ b/query.go
@@ -331,6 +331,12 @@ func (q *Query) End(c Cursor) *Query {
 	return q
 }
 
+// ENFABRICA PATCH
+// Export toProto for use in unit tests
+func (q *Query) ToProto(req *pb.RunQueryRequest) error {
+  return q.toProto(req)
+}
+
 // toProto converts the query to a protocol buffer.
 func (q *Query) toProto(req *pb.RunQueryRequest) error {
 	if len(q.projection) != 0 && q.keysOnly {
