diff --git a/apt/private/deb_import.bzl b/apt/private/deb_import.bzl
index 6ff7995..738a3a7 100644
--- a/apt/private/deb_import.bzl
+++ b/apt/private/deb_import.bzl
@@ -9,7 +9,7 @@ _RECOMPRESS_CMD = "$(ZSTD_BIN) -f --decompress $< --stdout | $(ZSTD_BIN) - --for
 genrule(
     name = "data",
     srcs = glob(["data.tar.*"]),
-    outs = ["data.tar.gz"],
+    outs = ["data_recompressed.tar.gz"],
     cmd = _RECOMPRESS_CMD,
     toolchains = ["@zstd_toolchains//:resolved_toolchain"],
     visibility = ["//visibility:public"],
