diff --git a/absl/copts/GENERATED_AbseilCopts.cmake b/absl/copts/GENERATED_AbseilCopts.cmake
index 430916f7..ac4c56f1 100644
--- a/absl/copts/GENERATED_AbseilCopts.cmake
+++ b/absl/copts/GENERATED_AbseilCopts.cmake
@@ -222,8 +222,3 @@ list(APPEND ABSL_RANDOM_HWAES_ARM64_FLAGS
 
 list(APPEND ABSL_RANDOM_HWAES_MSVC_X64_FLAGS
 )
-
-list(APPEND ABSL_RANDOM_HWAES_X64_FLAGS
-    "-maes"
-    "-msse4.1"
-)
diff --git a/absl/copts/GENERATED_copts.bzl b/absl/copts/GENERATED_copts.bzl
index 011d8a98..0073cff6 100644
--- a/absl/copts/GENERATED_copts.bzl
+++ b/absl/copts/GENERATED_copts.bzl
@@ -223,8 +223,3 @@ ABSL_RANDOM_HWAES_ARM64_FLAGS = [
 
 ABSL_RANDOM_HWAES_MSVC_X64_FLAGS = [
 ]
-
-ABSL_RANDOM_HWAES_X64_FLAGS = [
-    "-maes",
-    "-msse4.1",
-]
diff --git a/absl/copts/configure_copts.bzl b/absl/copts/configure_copts.bzl
index ca5f26da..684ca80d 100644
--- a/absl/copts/configure_copts.bzl
+++ b/absl/copts/configure_copts.bzl
@@ -18,7 +18,6 @@ load(
     "ABSL_RANDOM_HWAES_ARM32_FLAGS",
     "ABSL_RANDOM_HWAES_ARM64_FLAGS",
     "ABSL_RANDOM_HWAES_MSVC_X64_FLAGS",
-    "ABSL_RANDOM_HWAES_X64_FLAGS",
 )
 
 ABSL_DEFAULT_COPTS = select({
@@ -46,11 +45,8 @@ ABSL_DEFAULT_LINKOPTS = select({
 # environment to build an accelerated RandenHwAes library.
 ABSL_RANDOM_RANDEN_COPTS = select({
     # APPLE
-    ":cpu_darwin_x86_64": ABSL_RANDOM_HWAES_X64_FLAGS,
-    ":cpu_darwin": ABSL_RANDOM_HWAES_X64_FLAGS,
     ":cpu_x64_windows_msvc": ABSL_RANDOM_HWAES_MSVC_X64_FLAGS,
     ":cpu_x64_windows": ABSL_RANDOM_HWAES_MSVC_X64_FLAGS,
-    ":cpu_k8": ABSL_RANDOM_HWAES_X64_FLAGS,
     ":cpu_ppc": ["-mcrypto"],
     ":cpu_aarch64": ABSL_RANDOM_HWAES_ARM64_FLAGS,
 
diff --git a/absl/copts/copts.py b/absl/copts/copts.py
index e6e11949..4aa4b9e2 100644
--- a/absl/copts/copts.py
+++ b/absl/copts/copts.py
@@ -183,9 +183,5 @@ COPT_VARS = {
     # to improve performance of some random bit generators.
     "ABSL_RANDOM_HWAES_ARM64_FLAGS": ["-march=armv8-a+crypto"],
     "ABSL_RANDOM_HWAES_ARM32_FLAGS": ["-mfpu=neon"],
-    "ABSL_RANDOM_HWAES_X64_FLAGS": [
-        "-maes",
-        "-msse4.1",
-    ],
     "ABSL_RANDOM_HWAES_MSVC_X64_FLAGS": [],
 }
