diff --git a/bazel/deps.bzl b/bazel/deps.bzl
index 571dedf..b7207d2 100644
--- a/bazel/deps.bzl
+++ b/bazel/deps.bzl
@@ -10,9 +10,11 @@ load(
 )
 
 def jsonnet_go_dependencies(go_sdk_version = "host"):
-    go_rules_dependencies()
-    go_register_toolchains(version = go_sdk_version)
-    gazelle_dependencies()
+    # ENFABRICA PATCH
+    # These get loaded earlier; no need to load here
+    #go_rules_dependencies()
+    #go_register_toolchains(version = go_sdk_version)
+    #gazelle_dependencies()
     go_repository(
         name = "com_github_davecgh_go_spew",
         importpath = "github.com/davecgh/go-spew",
