diff --git a/bazel/cython_library.bzl b/bazel/cython_library.bzl
index 8e003c2246..ec37c68a54 100644
--- a/bazel/cython_library.bzl
+++ b/bazel/cython_library.bzl
@@ -60,8 +60,8 @@ def pyx_library(name, deps = [], py_deps = [], srcs = [], **kwargs):
             # Optionally use PYTHON_BIN_PATH on Linux platforms so that python 3
             # works. Windows has issues with cython_binary so skip PYTHON_BIN_PATH.
             cmd =
-                "PYTHONHASHSEED=0 $(location @cython//:cython_binary) --cplus $(SRCS) --output-file $(OUTS)",
-            tools = ["@cython//:cython_binary"] + pxd_srcs,
+                "PYTHONHASHSEED=0 $(location @cython//:cython) --cplus $(SRCS) --output-file $(OUTS)",
+            tools = ["@cython//:cython"] + pxd_srcs,
         )
 
     shared_objects = []
@@ -71,7 +71,7 @@ def pyx_library(name, deps = [], py_deps = [], srcs = [], **kwargs):
         native.cc_binary(
             name = shared_object_name,
             srcs = [stem + ".cpp"],
-            deps = deps + ["@local_config_python//:python_headers"],
+            deps = deps + ["@python3_11//:python_headers"],
             linkshared = 1,
         )
         shared_objects.append(shared_object_name)
