diff --git py/private/py_pex_binary.bzl py/private/py_pex_binary.bzl
index 9ee01fe..6998fbb 100644
--- py/private/py_pex_binary.bzl
+++ py/private/py_pex_binary.bzl
@@ -123,11 +123,18 @@ def _py_python_pex_impl(ctx):
     )
 
     return [
+        binary[PyInfo],
         DefaultInfo(files = depset([output]), executable = output),
     ]
 
 _attrs = dict({
-    "binary": attr.label(executable = True, cfg = "target", mandatory = True, doc = "A py_binary target"),
+    "binary": attr.label(
+        executable = True, 
+        cfg = "target", 
+        mandatory = True,
+        providers = [PyInfo],
+        doc = "A py_binary target",
+    ),
     "inject_env": attr.string_dict(
         doc = "Environment variables to set when running the pex binary.",
         default = {},
