diff --git a/BUILD b/BUILD
index 65e0cdc2e..0d0593b50 100644
--- a/BUILD
+++ b/BUILD
@@ -101,6 +101,10 @@ posix_copts = [
     "-Wwrite-strings",
     "-Wshadow",
     "-fno-common",
+    # no-pedanitc was added to fix below compilation errors.
+    # "ISO C does not support '__int128' types [-Werror=pedantic]"
+    # "ISO C does not allow extra ';' outside of a function [-Werror=pedantic]"
+    "-Wno-pedantic",
 
     # Modern build environments should be able to set this to use atomic
     # operations for reference counting rather than locks. However, it's
diff --git a/src/crypto/hrss/asm/poly_rq_mul.S b/src/crypto/hrss/asm/poly_rq_mul.S
index c37d7d0b4..0e2cf0620 100644
--- a/src/crypto/hrss/asm/poly_rq_mul.S
+++ b/src/crypto/hrss/asm/poly_rq_mul.S
@@ -12,7 +12,8 @@
 // OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 // CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-#if !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_SMALL) && defined(__linux__)
+// Code change to move the declarations under defined(__x86_64__) was picked up from origin/master-with-bazel branch.
+#if !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_SMALL) && defined(__linux__) && defined(__x86_64__)
 
 #if defined(BORINGSSL_PREFIX)
 #include <boringssl_prefix_symbols_asm.h>
